### PR TITLE
Solved boolean ambiguity for variables and tensors which contain one value.

### LINKED ISSUE #3652: Fix far-fetched Boolean ambiguity for byte tensor with one value in it.
Testing non-empty tensor in an if statement will unconditionally raise a RuntimeError stating that "bool value of non-empty torch.ByteTensor objects is ambiguous". But when there's only one value, it's bool value is pretty straightforward.
What do you think?

### PR DESCRIPTION
**Pull request by VladyslavZavadskyi**
Corrected according exception messages. Fixed some linter's code style warnings.
Resolves #3652.

### CODE REVIEW

File: torch/autograd/variable.py [hunk 1]
```diff
@@ -122,8 +122,11 @@ def __repr__(self):
     def __bool__(self):
         if self.data.numel() == 0:
             return False
-        raise RuntimeError("bool value of Variable objects containing non-empty " +
-                           torch.typename(self.data) + " is ambiguous")
+        elif self.numel() == 1:
+            return torch.squeeze(self.data)[0] != 0
```
Comment (apaszke) [inline torch/autograd/variable.py]: I think it's better to not repeat yourself and to do `self.data.__bool__()`

**Review by apaszke — CHANGES_REQUESTED** [review]
This looks good, but please revert all the changes that are not relevant to fixing the issue (formatting changes). They are correct, because we accept lines up to 120 chars.

File: torch/autograd/variable.py (hunk 1 above)
Comment (VladislavZavadskyy) [reply torch/autograd/variable.py]: Agreed, thanks.

**Comment by VladislavZavadskyy** [comment]
@apaszke check it out, please

**Review by apaszke — APPROVED** [review]
I changed "containing more than one object" to "more than one value". Looks good, thank you!

**Comment by soumith** [comment]
thank you @VladislavZavadskyy !

**Pull request merged**